### DISCUSSION TITLE
Joint Stereo vs Stereo

### DISCUSSION THREAD

**Discussion by LWinterberg** [audacity/audacity#4940] [Development]
We've taken the decision to remove the option for stereo vs joint stereo. **In short, there is no benefit to use "stereo" anymore and for typical stereo content, "joint stereo" preserves the quality better**. <sup>[[1]](https://wiki.hydrogenaud.io/index.php?title=Joint_stereo#MP3)</sup> Audacity will always export MP3s as Joint Stereo going forward.

In long: 

Back in the 90s and early 2000s, "joint stereo" meant "the entire file is encoded as M/S". Unlike normal stereo, where the two channels correspond to the left and right speakers, M/S stands for Mid/Side, where the mid channel takes the commonalities between the left and right channels, while the side channel takes their differences. For stereo files in which the left and right channel are largely similar (as is the case in most music), this lets the mp3 encoder (called LAME) encode the file a little bit more efficiently. In the case of the left and right channels being identical, normal stereo would have two full copies of it, while joint stereo would only have one copy in the Mid channel, with a bunch of nothing in the Side channel - offering the same quality at half the size, or doubling the quality at the same size.

However, the thinking continued, for content in which the channels *aren't* similar, what might happen is that the Side channel doesn't get enough bitrate, which would make the quality worse. And while that's true, LAME's "Joint Stereo" mode nowadays intelligently determines when to encode in M/S mode, and when to switch to L/R mode, which automatically ensures that it always delivers the highest quality at every given bitrate target.

If you want to use L/R stereo despite all this you can install a LAME executable separately, add it to PATH, and then export your file in Audacity via the (external program) option as the file format. `lame -m s - "%f"` is the command to export as forced L/R stereo, other modes can be found in the documentation below. 

<details><summary>relevant section of the LAME documentation</summary>

```
=======================================================================
Modes:
=======================================================================
-m m        mono
-m l	    get only the left channel of a stereo signal for a mono output
-m r	    get only the right channel of a stereo signal for a mono output
-m s        (forced) L/R stereo
-m j        joint stereo
-m f        forced mid/side stereo
-m d        dual (independent) channels. Its purpose was meant for dual language
            streams where only one of them should be decoded.
			Most decoders just decode them as a stereo stream.
-m a        Currently, a synonym of m j. (In older releases it selected
            different modes depending on the bitrate)

MONO is the default mode for mono input files.  If "-m m" is specified
for a stereo input file, the two channels will be averaged into a mono
signal.  (Note: See comments about the -a switch for RAW PCM streams)

(FORCED) L/R STEREO encodes the left and the right signals independently,
and gives more or less bits to each, depending on the currently available.

JOINT STEREO is the default mode of encoding.
jstereo means the encoder can use (on a frame by frame basis) either
L/R stereo or mid/side stereo.  In mid/side stereo, the mid(L+R) and side(L-R)
channels are encoded, and more bits are allocated to the mid channel
than the side channel.  When there isn't too much stereo separation, this
effectively increases the bandwidth, so having higher quality with the same
amount of bits.

Using mid/side stereo inappropriately can result in audible
compression artifacts.  Too much switching between mid/side and L/R
stereo can also sound bad. To determine when to switch to mid/side
stereo, LAME uses a much more sophisticated algorithm than that
described in the ISO documentation.

FORCED MID/SIDE STEREO forces all frames to be encoded mid/side stereo.  It
should only be used if you are sure every frame of the input file
has very little stereo seperation.

DUAL CHANNEL mode is similar to encode the left and right as two mono signals.
Its purpose was meant for Dual language streams where only one of them should
be decoded. Most decoders just decode them as a stereo stream

```

* Full LAME documentation: https://svn.code.sf.net/p/lame/svn/trunk/lame/USAGE
* recommended options: https://wiki.hydrogenaud.io/index.php/LAME

</details>

**Comment by petersampsonaudacity** [comment]
@crsib @LWinterberg 

This recent article indicates that **Stereo** is better for 256kbps (iTunes default rate) and above - and **Joint Stereo** is better for lower than 256kbps:
https://riverside.fm/blog/joint-stereo-vs-stereo

It also says
>stereo may produce a more high-fidelity and natural soundscape than joint stereo

So given that I export MP3s at 256kbps I really regret the loss of proper stereo with the new export.

Let's remember that joint stereo and lower bitrates were designed for a time when disk space was smaller and more expensive - and this no longer remains the case today.

----------------------------------------------------------------------------

And this article says:
https://audioviser.com/joint-stereo-vs-stereo/

>While Joint Stereo provides a more efficient use of storage and bandwidth, it can result in a slight loss of sound quality compared to Stereo. The encoding proces used in Joint Stereo can lead to audible artifacts, such as phase shifts and spectral imbalance, that can impact the overall sound quality of the audio.
>
>Stereo recordings, on the other hand, provide a higher level of sound quality as each channel is recorded separately, alowing for a more accurate representation of the original sound field.

**Reply by LWinterberg** [reply]
The important point here is that LAME's "Joint Stereo" really is "auto". It will pick between the two modes in order to achieve the best quality. The articles meanwhile are talking about either the mode LAME calls "Forced Joint Stereo" which we aren't using, or other kinds of joint stereo which LAME doesn't even support. 

It's also worth noting that Joint Stereo isn't inherently worse for all content. In fact, for most music, it's better. At 256 kbps, the forced L/R stereo mode effectively gives you 2 mono tracks, both at 128 kbps. In forced M/S mode, you can in theory do the same (2 mono tracks where one is getting the L+R signal, and the other is getting the L-R signal, both at 128 kbps). But in practice, music tends to have a lot of commonalities in both L and R channels; having a song where the singer is only audible on the left ear and the guitar is only audible in the right ear are pretty uncommon by and large. That means it's practical to give the mid channel more bits, so you have the mid channel with eg. 200 kbps and the side channel with just 56 kbps, which lets you have a higher quality overall. In the extreme case, where both L and R channels are identical, M/S can give 256 kbps to the mid channel and 0 to the side, which gives forced joint stereo twice the quality to forced L/R stereo! 

> phase shifts and spectral imbalance

These specific artefacts occur with a different kind of Joint Stereo, called Intensity Stereo, or the even more advanced Parametric Stereo. They employ some psychoacoustic trickery I don't understand, but the end result is very efficient but lossy in those areas. LAME does not support these techniques at all, it only does M/S and L/R stereo. 

To convert between the two, here's all the math required: 

```
Encoding
M = (L + R) / 2, 
S = (L - R) / 2
Decoding
L = M + S, 
R = M - S
```

The conversion between L/R and M/S is completely lossless and doesn't incur any phase issues or spectral issues whatsoever.

**Comment by Scribliotech** [comment]
Note that the Audacity manual still shows (incorrectly) that this is an option. Note the first paragraph suggest that Joint Stereo is an option:
https://manual.audacityteam.org/man/mp3_export_options.html

![image](https://github.com/audacity/audacity/assets/7894356/6591c3d3-7e12-4241-8565-4aa2bfe171b7)

I was quite confused by the removal of the choice from the Export dialog in the new version and that it no longer offers an option for Joint Stereo. I spent quite a while trying to figure out how to get Joint Stereo before realizing that's now the default export format.

**Comment by petersampsonaudacity** [comment]
@Scribliotech 

Yes this was also reported on the Forum - nut thanks for your report here.

I have foxed this in the alpha manual for the upcoming 3.5.0 release.
